CHANGES/5853.bugfix

```diff
@@ -0,0 +1 @@
+Don't cancel current task when entering a cancelled timer.
```

Comment:
This looks like a feature or even a breaking change: you literally revert what the current tests expect.

Reply:
True. Unfortunately I could see no rationale or explanation in the code or commit/PR which introduced the change (just 'refactoring'). The only thing I can imagine is that someone got mixed up with the cancelling behaviour used in `.timeout()` which aborts the `with` block early and then is suppressed in the `__exit__()`.

But, cancelling here is totally different and actually cancels whatever task the user was running (in the linked issue, that is the entire application).

Reply:
Yep, I saw that. I just don't know what Andrew had in mind for this.